helper/memory_calculator.go

```diff
@@ -103,7 +106,25 @@ func (m MemoryCalculator) Execute() (map[string]string, error) {
 	}
 
 	if totalMemory == UnsetTotalMemory {
-		m.Logger.Info("WARNING: Container memory limit unset. Configuring JVM for 1G container.")
+		if b, err := ioutil.ReadFile(m.MemoryInfoPath); err != nil && !os.IsNotExist(err) {
+			m.Logger.Info("Unable to read %s\n%w", m.MemoryInfoPath, err)
+		} else if err == nil {
+			rp := regexp.MustCompile("MemAvailable:\\s*(\\d*)\\skB")
+			m := rp.FindStringSubmatch(string(b))
+
+			if m != nil && len(m) > 1 {
```

Comment:
```suggestion
			if len(m) > 1 {
```
`len` of a `nil` slice is `0` so this is safe.